profiles/serializers_test.py

```diff
@@ -304,3 +309,158 @@ def test_delete_employment(self):
         assert employment3.profile.work_history.count() == 1
+
+
+class ProfileFilledOutTests(ESTestCase):
+    """Tests for validating filled out profiles"""
+
+    def create_profile(self):
+        """
+        Create a profile and social auth
+        """
+        with mute_signals(post_save):
+            profile = ProfileFactory.create(
+                filled_out=True,
+                agreed_to_terms_of_service=True,
+            )
+            EmploymentFactory.create(profile=profile)
+            EducationFactory.create(profile=profile)
+            profile.user.social_auth.create(
+                provider=EdxOrgOAuth2.name,
+                uid="{}_edx".format(profile.user.username)
+            )
+            return profile
+
+    def test_success(self):
+        """
+        Test a successful validation
+        """
+        profile = self.create_profile()
+
+        serializer = ProfileFilledOutSerializer(profile)
+        data = serializer.to_representation(profile)
+        ProfileFilledOutSerializer(data=data).is_valid(raise_exception=True)
+
+    def test_filled_out_false(self):
+        """
+        filled_out cannot be set to false
+        """
+        profile = self.create_profile()
+
+        serializer = ProfileFilledOutSerializer(profile)
+        data = serializer.to_representation(profile)
+        data['filled_out'] = False
+        with self.assertRaises(ValidationError) as ex:
+            ProfileFilledOutSerializer(data=data).is_valid(raise_exception=True)
+        assert ex.exception.args[0] == {'non_field_errors': ['filled_out cannot be set to false']}
+
+    def test_agreed_to_terms_of_service_false(self):
+        """
+        agreed_to_terms_of_service cannot be set to false
+        """
+        profile = self.create_profile()
+
+        serializer = ProfileFilledOutSerializer(profile)
+        data = serializer.to_representation(profile)
+        data['agreed_to_terms_of_service'] = False
+        with self.assertRaises(ValidationError) as ex:
+            ProfileFilledOutSerializer(data=data).is_valid(raise_exception=True)
+        assert ex.exception.args[0] == {'non_field_errors': ['agreed_to_terms_of_service cannot be set to false']}
+
+    def test_required_fields(self):
+        """
+        All fields are required after a profile is marked filled_out
+        """
+        profile = self.create_profile()
+
+        serializer = ProfileFilledOutSerializer(profile)
+        data = serializer.to_representation(profile)
+        # skip fields that are generated, and work_history and education which are tested elsewhere
+        keys = set(ProfileFilledOutSerializer.Meta.fields) - {
+            'username',
+            'profile_url_full',
+            'profile_url_large',
+            'profile_url_medium',
+            'profile_url_small',
+            'pretty_printed_student_id',
+            'work_history',
+            'education',
+        }
+        for key in keys:
```

Comment:
this loop is repeated with some minor changes in test_work_history and test_education. would strongly prefer that we abstract it - they are testing the same general idea, after all. i'll think about a good way to go about it